Sources/Orbit/Support/Components/TimelineIndicator.swift

```diff
@@ -0,0 +1,89 @@
+                icon
+            case .current(.warning), .current(.critical), .current(.success):
+                Circle()
+                    .frame(
+                        width: (animationLoopTrigger ? .medium : .xMedium) * sizeCategory.ratio,
+                        height: (animationLoopTrigger ? .medium : .xMedium) * sizeCategory.ratio
+                    )
+                    .foregroundColor(animationLoopTrigger ? Color.clear : type.color.opacity(0.1))
+                    .animation(animation, value: animationLoopTrigger)
+                    .overlay(icon)
+            case .past:
+                Circle()
+                    .foregroundColor(type.color.opacity(0.1))
+                    .frame(width: .xMedium * sizeCategory.ratio, height: .xMedium * sizeCategory.ratio)
+                    .overlay(icon)
+        }
+    }
+
+    @ViewBuilder var icon: some View {
+        switch type {
+            case .future:
+                Circle()
+                    .strokeBorder(type.color, lineWidth: 2)
+                    .background(Circle().fill(Color.whiteNormal))
+                    .frame(width: .small * sizeCategory.ratio, height: .small * sizeCategory.ratio)
+            case .current(.info):
+                ZStack(alignment: .center) {
+                    Circle()
+                        .frame(
+                            width: (animationLoopTrigger ? .medium : .xMedium) * sizeCategory.ratio,
+                            height: (animationLoopTrigger ? .medium : .xMedium) * sizeCategory.ratio
+                        )
+                        .animation(animation, value: animationLoopTrigger)
+                        .foregroundColor(type.color.opacity(0.1))
+
+                    Circle()
+                        .strokeBorder(type.color, lineWidth: 2)
+                        .background(Circle().fill(Color.whiteNormal))
+                        .frame(width: .small * sizeCategory.ratio, height: .small * sizeCategory.ratio)
+
+                    Circle()
+                        .frame(width: .xxSmall * sizeCategory.ratio, height: .xxSmall * sizeCategory.ratio)
+                        .scaleEffect(animationLoopTrigger ? .init(width: 0.5, height: 0.5) : .init(width: 1, height: 1))
+                        .foregroundColor(animationLoopTrigger ? Color.clear : type.color)
+                        .animation(animation, value: animationLoopTrigger)
+
+                }
+        case .current(.warning), .current(.critical), .current(.success), .past:
+                Icon(type.iconSymbol, size: .small, color: type.color)
+                    .background(Circle().fill(Color.whiteNormal).padding(2))
+        }
+    }
+
+    var animation: Animation {
+        Animation.easeInOut.repeatForever().speed(0.25)
+    }
+
+    func iconContentBaselineOffset(height: CGFloat) -> CGFloat {
+        Icon.Size.small.textBaselineAlignmentGuide(sizeCategory: sizeCategory, height: height)
+    }
```

Comment:
If we will need to do Status-related adjustments anyway, it would be nice to have a preview for this component as well (to see all those possible states)

Reply:
I wouldn't do it for the same reason why I wouldn't do it for `TimelineLine` – they are both `internal` and their states are still tested anyways via snapshots of `Timeline` and `TimelineItem`. In `TimelineItem` you can see all the possible states of the indicator already. what do you think?